runtime/client_http_response_test.go

```diff
@@ -230,6 +242,79 @@ func TestUnknownStatusCode(t *testing.T) {
 	assert.Len(t, logs["Finished an outgoing client HTTP request"], 1)
 }
 
+func TestNotFollowRedirect(t *testing.T) {
+	gateway, err := benchGateway.CreateGateway(
+		defaultTestConfig,
+		&testGateway.Options{
+			LogWhitelist: map[string]bool{
+				"Could not emit statusCode metric": true,
+			},
+			KnownHTTPBackends: []string{"bar"},
+			ConfigFiles:       util.DefaultConfigFiles("example-gateway"),
+		},
+		exampleGateway.CreateGateway,
+	)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer gateway.Close()
+
+	bgateway := gateway.(*benchGateway.BenchGateway)
+
+	addr := bgateway.HTTPBackends()["bar"].RealAddr
+	baseURL := "http://" + addr
+
+	redirectURI := baseURL + "/bar-path"
+	fakeEcho := func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Location", redirectURI)
+		w.WriteHeader(303)
+		_, err := w.Write([]byte(`false`))
+		assert.NoError(t, err)
+	}
+
+	fakeNormal := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(999)
+		_, err := w.Write([]byte(`false`))
+		assert.NoError(t, err)
+	}
+
+	bgateway.HTTPBackends()["bar"].HandleFunc("POST", "/bar/echo", fakeEcho)
+	bgateway.HTTPBackends()["bar"].HandleFunc("GET", "/bar-path", fakeNormal)
```

Comment:
this doesn't seem to be used, remove it.

Reply:
make sense, since we don't follow the redirect, we don't need it anymore.